aea/crypto/cosmos.py

```diff
@@ -461,16 +463,90 @@ def _try_get_state(  # pylint: disable=unused-argument
             result = response.json()
         return result
 
-    def get_deploy_transaction(  # pylint: disable=arguments-differ
+    def get_deploy_transaction(
+        self, contract_interface: Dict[str, str], deployer_address: Address, **kwargs
+    ) -> Optional[JSONLike]:
+        """
+        Get the transaction to deploy the smart contract.
+
+        Dispatches to _get_storage_transaction and _get_init_transaction based on kwargs.
+
+        :param contract_interface: the contract interface.
+        :param deployer_address: The address that will deploy the contract.
+        :returns tx: the transaction dictionary.
+        """
+        denom = (
+            kwargs.pop("denom") if kwargs.get("denom", None) is not None else self.denom
+        )
+        chain_id = (
+            kwargs.pop("chain_id")
+            if kwargs.get("chain_id", None) is not None
+            else self.chain_id
+        )
+        account_number, sequence = self._try_get_account_number_and_sequence(
+            deployer_address
+        )
+        if account_number is None or sequence is None:
+            return None
+        label = kwargs.pop("label", None)
+        code_id = kwargs.pop("code_id", None)
+        amount = kwargs.pop("amount", None)
+        init_msg = kwargs.pop("init_msg", None)
+        unexpected_keys = [
+            key for key in kwargs.keys() if key not in ["tx_fee", "gas", "memo"]
+        ]
+        if len(unexpected_keys) != 0:
+            raise ValueError(f"Unexpected keyword arguments: {unexpected_keys}")
+        if label is None and code_id is None and amount is None and init_msg is None:
+            return self._get_storage_transaction(
+                contract_interface,
+                deployer_address,
+                denom,
+                chain_id,
+                account_number,
+                sequence,
+                **kwargs,
+            )
+        if label is None:
+            raise ValueError(
+                "Missing required keyword argument `label` of type `str` for `_get_init_transaction`."
+            )
+        if code_id is None:
+            raise ValueError(
+                "Missing required keyword argument `code_id` of type `int` for `_get_init_transaction`."
+            )
+        if amount is None:
+            raise ValueError(
+                "Missing required keyword argument `amount` of type `int` for `_get_init_transaction`."
+            )
+        if init_msg is None:
+            raise ValueError(
+                "Missing required keyword argument `init_msg` of type `JSONLike` `for `_get_init_transaction`."
+            )
+        return self._get_init_transaction(
+            deployer_address,
+            denom,
+            chain_id,
+            account_number,
+            sequence,
+            amount,
+            code_id,
+            init_msg,
+            label,
+            **kwargs,
```

Comment:
just a point to think about.

probably introduce an internal class for transaction data, it will validate and store all the values, so no need to pass a lot of arguments?